### PR TITLE
Remove artifact registry module from GCP conformance CI

### PR DESCRIPTION
**Pull request by roger2hk**
Towards #53.

### CODE REVIEW

File: deployment/live/gcp/ci/README.md [hunk 1]
```diff
@@ -30,6 +30,8 @@ export GOOGLE_REGION={VALUE} # e.g: us-central1
 unset TESSERA_BASE_NAME
 ```
 
+Add the Artifact Registry which is in the Cloud Build pull request. The expected Docker repository is `${GOOGLE_REGION}-docker.pkg.dev/${GOOGLE_PROJECT}/docker-ci`. (The Artifact Registry terraform module is ready at [/deployment/modules/gcp/artifactregistry/](/deployment/modules/gcp/artifactregistry/).)
```
Comment (phbnf) [inline deployment/live/gcp/ci/README.md:33]: Do you want to point to a PR  / directory? Or maybe just the artifact registry module? You could say that these instructions expect for an Aritfact Registry to exist, which can be setup manually, or using the terraform module we provide.

File: deployment/live/gcp/ci/README.md
Comment (phbnf) [inline deployment/live/gcp/ci/README.md:1]: "define a CI environment to run the SCTFE, backed by Trillian Tessera.", maybe we could say "define a CI environment to run the SCTFE on CloudRun, backed by Trillian Tessera."

Comment (phbnf) [inline deployment/live/gcp/ci/README.md:1]: We could also add a section with "automated deployment using terraform", pointing at the cloudbuild terraform

Comment (roger2hk) [reply deployment/live/gcp/ci/README.md:1]: Done.

Comment (roger2hk) [reply deployment/live/gcp/ci/README.md:1]: Done.

File: deployment/live/gcp/ci/README.md (hunk 1 above)
Comment (roger2hk) [reply deployment/live/gcp/ci/README.md:37]: This is the step for manual deployment, so it is expect to setup manually. Using the terraform module requires more effort than setting it up manually but the module itself is useful to tell what to setup.

Comment (phbnf) [reply deployment/live/gcp/ci/README.md]: Great! If we point people at a specific Pull request, let's point them at it? Agreed with you that an example is worth many explanations. If it's just an example, let's not mandate that people do it.

I'm thinking of something like:

> You need an Artifact Registry repository to store container images; adapt the configs and commands below to use your registry of choice. The rest of these instructions assume that the repository is hosted on GCP, and called `${GOOGLE_REGION}-docker.pkg.dev/${GOOGLE_PROJECT}/docker-ci`. For reference, [here's a terraform module](/deployment/modules/gcp/artifactregistry/) you can use to set up such a registry.

Comment (roger2hk) [reply deployment/live/gcp/ci/README.md]: Done.

**Review by phbnf — APPROVED** [review]

**Pull request merged**